covid19-dashboard/src/App.tsx

```diff
@@ -261,17 +267,42 @@ class App extends React.Component<AppPropsType, AppStateType> {
     // Example: parent places of Florida: ["World", "US", "Florida"].
     const parentPlaces = this.getParentPlaces();
 
+    // Get the latest date for all places.
+    const dates = places
+      .map(place => {
+        const keyToTimeSeries = place.keyToTimeSeries[Config.tableDefaultSortBy] || {};
+        // Config.defaultKey contains our reference key for dates.
+        // AKA, what is our most complete dataset? We want to get the date
+        // from that dataset.
+        const dates = Object.keys(keyToTimeSeries);
+        return getLatestDate(dates);
+      })
+      .filter(date => date);
+
+    // From all the latest dates, calculate the latest date.
+    // This will tell us the actual date.
+    // Example: If Spain was updated August 17 and the US on August 18,
+    // Our latest date will be "August 18" because, that's the most recent date.
+    const maxDate = getLatestDate(dates);
+
+    // Format maxDate into a more readable version.
+    // Only format the date, if maxDate is valid.
+    let formattedDate = '';
+    if (maxDate) {
+      formattedDate = dayjs(maxDate).format('MMM D, YYYY');
+    }
+
     // Text describing the types of places viewing viewed.
     // "Countries in", "States in", "Counties in".
     const pluralPlaceTypes: {[key: string]: string} = Config.pluralPlaceTypes;
     let subtitle = '';
 
-
     // If placeName is "" or undefined, don't display a subtitle.
     // It means hasn't finished loading.
     if (selectedPlace?.name) {
+      const pluralPlaceType = pluralPlaceTypes[this.placeTypeToShow];
```

Comment:
consider adding a check somewhere that all place types are in the config.

Reply:
fixed.